palanaeum/static/palanaeum/js/page_nav.js

```diff
@@ -49,4 +49,14 @@ $(function(){
             $("#desktop-header-small:visible").fadeOut("fast");
         }
     });
-});
\ No newline at end of file
+});
+
+function show_hidden_pages(elipsis_element) {
+    elipsis_element.classList.add("hidden-pg-num");
```

Comment:
This whole operation can be done faster using jQuery, without the need to iterate over all the elements:

```
# Get all .hidden-pg-num elements and remove the class from them
$(".hidden-pg-num").removeClass("hidden-pg-num");
# Re-add the class to the ellipsis element.
elipsis_element.addClass("hidden-pg-num");
```

or you can just use `.hide()` and `.show()`

Reply:
I figured jQuery had something like this but I haven't used it before. But I went with this method so that it would only expand the ellipsis that was clicked. Because the page nav bar is in three places, the page will either have 0, 3, or 6 ellipsis elements. If there are a lot of hidden pages and we showed them in all 6 places, it could push down the content at the bottom of the page, which would be annoying if you wanted to click one of the newly-revealed page numbers. So I decided to just expand the pages of that one ellipsis. Hence passing in the DOM element that was clicked and traversing one-at-a-time to the right.

Reply:
Okay it actually looks like the recents page that nav bar only once, but the search page has it three times.